### ISSUE TITLE
`fmt: on/off` pragmata around a class signature don't work properly

### ISSUE THREAD

**Issue by Xophmeister** [astral-sh/ruff#20425]
### Summary

If you put `fmt: off` and `fmt: on` pragmata around a class signature, the `fmt: on` reset pragma doesn't seem to be followed and the rest of the module is not formatted.

For example:

https://play.ruff.rs/0a9e3f0a-29a4-4774-81ab-fdeec065e636

The above is a minimal example, but I was using this to try to make my generics easier to read:

```python
# fmt: off
class ConcreteThingy(ThingyABC[
    Path,    # SourceT
    str,     # MetadataT
    SHA256,  # HashT
]):
# fmt: on

    # Nothing from this point is formatted...
    ...

# ...even outside the class definition
```

If you try putting the pragmata deeper into the tree -- say into the subclass parentheses, or the generics parentheses -- then the `fmt: off` pragma is completely ignored. For example:

Before:
```python
class ConcreteThingy(
    ThingyABC[
        # fmt: off
        Path,    # SourceT
        str,     # MetadataT
        SHA256,  # HashT
        # fmt: on
    ]
): ...
```

After:
```python
class ConcreteThingy(
    ThingyABC[
        # fmt: off
        Path,  # SourceT
        str,  # MetadataT
        SHA256,  # HashT
        # fmt: on
    ]
): ...
```

### Version

ruff 0.11.10

**Comment by ntBre** [comment]
Thanks for the report! It looks like we also don't flag [invalid-formatter-suppression-comment (RUF028)](https://docs.astral.sh/ruff/rules/invalid-formatter-suppression-comment/#invalid-formatter-suppression-comment-ruf028) anywhere here. 

[playground](https://play.ruff.rs/5b335c3c-f4d0-4e01-a1a4-16585333d871)

I found a somewhat related issue in https://github.com/astral-sh/ruff/issues/9875, and it looks like the suggestion to use `fmt: skip` instead may also help here:

```py
class ConcreteThingy(
    ThingyABC[
        Path,    # SourceT
        str,     # MetadataT
        SHA256,  # HashT
    ]
): # fmt: skip
    ...
```

**Comment by MichaReiser** [comment]
The formatter behavior is intentional. `fmt: off`/`fmt: on` comments are only allowed between statements:

> As such, adding # fmt: on and # fmt: off comments within expressions will have no effect. In the following example, both list entries will be formatted, despite the # fmt: off:
>
> https://docs.astral.sh/ruff/formatter/#format-suppression

For your use case, the recommended approach is to use `fmt: skip` as shown by @ntBre 

That RUF028 doesn't flag the invalid suppression comment is a bug.
